### PR TITLE
Add test for FEEvaluationShift with FE_Q and FE_DGQ

### PR DESCRIPTION
**Pull request by bergbauer**
Request from https://github.com/dealii/dealii/pull/14119#pullrequestreview-1033057735

### CODE REVIEW

File: tests/matrix_free/fe_q_fe_evaluation_shift.cc [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+  if (dim >= 1)
+    dealii::GridTools::collect_periodic_faces(tria, 0, 1, 0, periodic_faces);
+
+  if (dim >= 2)
+    dealii::GridTools::collect_periodic_faces(tria, 2, 3, 1, periodic_faces);
+
+  tria.add_periodicity(periodic_faces);
+
+  FE_Q<dim>       fe(fe_degree);
```
Comment (peterrum) [inline tests/matrix_free/fe_q_fe_evaluation_shift.cc:293]: You are using the class for `FE_DGQ`!? What about testing that here as well?

File: tests/matrix_free/fe_q_fe_evaluation_shift.cc [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+
+  MappingQ<dim> mapping(1);
+  QGauss<1>     quad(n_points);
+
+  AffineConstraints<Number> constraint;
```
[the remark this answers is not in the document]
Comment (bergbauer) [reply tests/matrix_free/fe_q_fe_evaluation_shift.cc]: I added constraints also for the DoFHandler!

**Review by peterrum — APPROVED** [review]

**Review by tamiko — APPROVED** [review]

File: tests/matrix_free/fe_evaluation_shift.cc [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+    this->interior_face = true;
+
+    if (shape_info != nullptr)
+      {
+        AssertDimension(shape_info->n_q_points, this->data->n_q_points);
+        AssertDimension(shape_info->dofs_per_component_on_cell,
+                        this->data->dofs_per_component_on_cell);
+        this->data = shape_info;
+      }
```
Comment (kronbichler) [inline tests/matrix_free/fe_evaluation_shift.cc:139]: You're not using this path of the code, do you? Do you want to extend the test to also make use of a `ShapeInfo` evaluated outside of the cell?

Comment (bergbauer) [reply tests/matrix_free/fe_evaluation_shift.cc:139]: Yes this is a good idea!

Comment (bergbauer) [reply tests/matrix_free/fe_evaluation_shift.cc:139]: Updated!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**